src/main/kotlin/com/hypto/iam/server/Application.kt

```diff
@@ -92,6 +96,15 @@ fun Application.handleRequest() {
                 }
             }
         }
+        apiKeyAuth("passcode-auth") {
+            validate { tokenCredential: TokenCredential ->
+                tokenCredential.value?.let {
+                    passcodeRepo.findById(it)?.let {
```

Comment:
<img width="727" alt="image" src="https://user-images.githubusercontent.com/98082541/168538916-f16840eb-62f7-4d4e-8d95-570e413cc32e.png">
As suggested in their website it uses some experimental APIs, Sould we go for it?

Reply:
This a documentation issue. The first version of DoubleReceive was using experimental api but in the latest version, experimental apis are not used. So we should be good to go.

Old version using exprimental api - https://github.com/ktorio/ktor/commit/6a428f050d8fd8537540783aeee3e1f3aa71fe2d

Latest version which is not exprimental api - https://github.com/ktorio/ktor/blob/main/ktor-server/ktor-server-plugins/ktor-server-double-receive/jvmAndNix/src/io/ktor/server/plugins/doublereceive/DoubleReceive.kt

Reply:
https://github.com/hwslabs/iam/blob/d0736df916b87a529c8d8e3d03646aa0ffeda1d9/src/main/kotlin/com/hypto/iam/server/service/OrganizationsService.kt#L54-L61

We are also validating the purpose and expiry here as well.

Reply:
> So can we validate the passcode and expiry status in the auth layer and validate the purpose in the service itself, so that we can use that same auth layer for forgot and invite user.

+1. This will simplify api code. Also we can avoid 2 db calls.